pkg/kubernetes/tls/certs.go

```diff
@@ -0,0 +1,102 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package tls
+
+import (
+	"context"
+	"time"
+
+	"github.com/devfile/devworkspace-operator/internal/cluster"
+	"github.com/devfile/devworkspace-operator/webhook/server"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type GenCertParams struct {
+	RequesterName string
+	Domain        string
+	Namespace     string
+	TLSSecretName string
+	CASecretName  string
+}
+
+// TLS related constants
+const (
+	TLSSelfSignedCertificateSecretName = "devworkspace-self-signed-certificate"
```

Comment:
I think it makes sense to make it up to client (webhooks server, workspace routing) to decide where CA should be stored.